components/cluster.py

```diff
@@ -143,9 +141,11 @@ def update_output(start_date, end_date, session_id):
     data = data.filter_by_dates(start_date, end_date)
 
     df = data.data
-    edges, df = compute_edges(data, df)
-    if edges == []:
+    edges = []
+    if not df.empty:
+        edges, df = compute_edges(data)
 
+    if edges == []:
         figures = [go.Figure()]
     else:
         figures = get_figures(edges, df)
```

Comment:
This logic is a bit weird in my opinion. Wouldn't something like this do the same?
```suggestion
    if df.empty:
        figures = [go.Figure()]
    else:
        edges, df = compute_edges(data)
        figures = get_figures(edges, df)
```